frontend/src/metabase/browse/containers/BrowseModels.tsx

```diff
@@ -0,0 +1,278 @@
+      // FIXME: Not sure that 'Model Click' is right; this is modeled on the database grid which has 'Database Click'
+      data-metabase-event={`${ANALYTICS_CONTEXT};Model Click`}
+    >
+      <ModelCard>
+        <h4 className="text-wrap" style={{ lineHeight: "16px" }}>
+          <MultilineEllipsified id={headingId}>
+            {model.name}
+          </MultilineEllipsified>
+        </h4>
+        <Text size="xs" style={{ height: "32px" }}>
+          <MultilineEllipsified
+            tooltipMaxWidth="100%"
+            className={model.description ? "" : "text-light"}
+          >
+            {model.description || "No description."}{" "}
+          </MultilineEllipsified>
+        </Text>
+        <LastEditInfoLabel
+          item={modelWithHistory}
+          fullName={modelWithHistory["last-edit-info"].full_name}
+          className={"last-edit-info-label-button"}
+          tooltipProps={{
+            label: tooltipLabel,
+            withArrow: true,
+          }}
+          // TODO: Use first name and last initial
+        >
+          {lastEditorName}
+          {lastEditorName && howLongAgo ? (
+            <LastEditedInfoSeparator>•</LastEditedInfoSeparator>
+          ) : null}
+          {howLongAgo}
+        </LastEditInfoLabel>
+      </ModelCard>
+    </Link>
+  );
+};
+
+/** Sort models firstly by collection name,
+ * secondly by collection id,
+ * thirdly by model name, and
+ * lastly by model id. */
+const sortModels = (a: SearchResult, b: SearchResult) => {
+  const fallbackSortValue = Number.MAX_SAFE_INTEGER;
+
+  // Sort first on the name of the model's parent collection, case insensitive
+  const collectionNameA =
+    a.collection?.name?.toLowerCase() || fallbackSortValue;
+  const collectionNameB =
+    b.collection?.name?.toLowerCase() || fallbackSortValue;
+
+  if (collectionNameA < collectionNameB) {
+    return -1;
+  }
+  if (collectionNameA > collectionNameB) {
+    return 1;
+  }
+
+  // If the two models' parent collections have the same name, sort on the id of the collection
+  const collectionIdA = a.collection?.id ?? fallbackSortValue;
+  const collectionIdB = b.collection?.id ?? fallbackSortValue;
+
+  if (collectionIdA < collectionIdB) {
+    return -1;
+  }
+  if (collectionIdA > collectionIdB) {
+    return 1;
+  }
+
+  const nameA = a.name.toLowerCase() || fallbackSortValue;
+  const nameB = b.name.toLowerCase() || fallbackSortValue;
+
+  // If the two collection ids are the same, sort on the names of the models
+  if (nameA < nameB) {
+    return -1;
+  }
+  if (nameA > nameB) {
+    return 1;
+  }
+
+  // If the two models have the same name, sort on id
+  const idA = a.id ?? fallbackSortValue;
+  const idB = b.id ?? fallbackSortValue;
+
+  if (idA < idB) {
+    return -1;
+  }
+  if (idA > idB) {
+    return 1;
+  }
+
+  return 0;
+};
+
+const CollectionHeader = ({
+  collection,
+  style,
+  id,
+}: {
+  collection?: Pick<Collection, "id" | "name"> | null;
```

Comment:
`Urls.collection` expects a `Collection`, but only a `Partial<Collection>` is available as the `collection` property of a model found through `/api/search`, so we need this weird intermediate type.